doc/cabal-project-description-file.rst

```diff
@@ -40,13 +40,28 @@ directories when there is none in the current directory.
 Conditionals and imports
 ------------------------
 
-As of ``cabal-install`` version 3.8, cabal supports conditional logic
-and imports in ``cabal.project`` files. :ref:`conditions` in cabal
-may case on operating system, architecture, and
-compiler (i.e. there is no support for a notion of custom flags in
-project files). Imports may specify local filepaths or remote urls,
-and may reference either cabal.project files or v1-style cabal.config
-freeze files. As a usage example:
+As of ``cabal-install`` version 3.8, cabal supports conditional logic and
+imports in ``cabal.project`` files.
+
+    .. warning::
+
+      While :ref:`conditional blocks<conditional-blocks>` can appear anywhere
+      within component or common sections of a package, their placement within a
+      project is restricted.  Conditions may only be introduced at the top level
+      of a project.
+
+      Projects have fewer :ref:`condition tests<conditions>` as only packages
+      can have tests for flags.  The remaining tests are available to projects,
```

Comment:
comma is wrong here. "The flags available to projects are operating system..." would be clearer.